viamonvif/service.go

```diff
@@ -377,6 +384,18 @@ func fetchImageFromRTSPURL(ctx context.Context, logger logging.Logger, rtspURL s
 	}
 }
 
+// generateUniqueName creates a unique name by adding timestamp and random bytes
+func generateUniqueName(prefix string) (string, error) {
+	randomBytes := make([]byte, rtspNameSaltLength)
+	if _, err := rand.Read(randomBytes); err != nil {
+		return "", err
+	}
+	timestamp := time.Now().UnixNano()
+	randomHex := hex.EncodeToString(randomBytes)
+	uniqueName := fmt.Sprintf("%s-%d-%s", prefix, timestamp, randomHex)
```

Comment:
I don't think we are going to get 64 calls in the same nanosecond. If we need to make the names unique for practical purposes, the nanosecond timestamp should be sufficient.

That also eliminates the need for this function to return an error.